pkg/collector/testdata/relabel_config_expected_with_sd_config.yaml

```diff
@@ -0,0 +1,57 @@
+exporters: null
+logging: null
+processors: null
+receivers:
+  prometheus:
+    config:
+      global:
+        evaluation_interval: 1m
+        scrape_interval: 1m
+        scrape_timeout: 10s
+      scrape_configs:
+      - honor_labels: true
+        http_sd_configs:
+        - url: http://test-targetallocator:80/jobs/service-x/targets?collector_id=$POD_NAME
+        job_name: service-x
+        metric_relabel_configs:
+        - action: keep
+          regex: (.*)
+          separator: ;
+          source_labels:
+          - label1
+        - action: labelmap
+          regex: (.*)
+          separator: ;
+          source_labels:
+          - label4
+        metrics_path: /metrics
+        relabel_configs:
+        - action: keep
+          regex: (.*)
+          source_labels:
+          - label1
+        - action: replace
+          regex: (.*)
+          replacement: $$1_$2
```

Comment:
should the `$2` be escaped as `$$2`? This would be what we tell a collector user to do, no?

Reply:
i.e. does the first `$` escape just `$1` or does it escape the entire `$1_$2`?

Reply:
@jaronoff97, the code path for constructing the collector configMap does not escape any dollar characters ('$') in the user-specified configuration. Instead, it updates the HTTP SD service configuration by directly modifying the promConfigMap, without the need for unmarshalling and updating the configuration. It is expected that users will properly escape '$' characters in their Prometheus configuration. This approach allows us to support environment variable substitution as well. For instance, in the example above, if a user entered "$$1_$2" in the prometheus configuration, the config map would be created with that value. The "$$1" part will be processed by the regular expression match, while the "$2" part will be replaced by the corresponding environment variable. The fix ensures that there are no problems when the users specify "$$" in the prometheus configuration (as per the OTel guideline).

Reply:
i see, so in this case `$$1_$2` for a regex match of 'thing' would be `thing_` if $2 is unset? I guess my question is, should we also escape the $2 in our test?